app/policies/source_policy.rb

```diff
@@ -0,0 +1,25 @@
+class SourcePolicy < ApplicationPolicy
+  def index?
+    user.admin?
+  end
+
+  def new?
+    user.admin?
+  end
+
+  def create?
+    user.admin?
+  end
+
+  def edit?
+    user.admin? && (user.organization_id == record.organization_id)
+  end
+
+  def update?
+    user.admin? && (user.organization_id == record.organization_id)
+  end
+
+  def destroy?
+    user.admin? && (record.id != user.id)
```

Comment:
do we need to check the id difference?

Reply:
I think it should protect admin from deleting himself. And it was one of requirements